internal/build/job/manager.go

```diff
@@ -7,59 +7,48 @@ import (
 
 	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/build"
-	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
 	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
-	batchv1 "k8s.io/api/batch/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-const jobHashAnnotation = "kmm.node.kubernetes.io/last-hash"
-
-var errNoMatchingBuild = errors.New("no matching build")
-
 type jobManager struct {
-	client client.Client
-	maker  Maker
-	helper build.Helper
+	client    client.Client
+	maker     Maker
+	helper    build.Helper
+	jobHelper utils.JobHelper
 }
 
-func NewBuildManager(client client.Client, maker Maker, helper build.Helper) *jobManager {
+func NewBuildManager(client client.Client, maker Maker, helper build.Helper, jobHelper utils.JobHelper) *jobManager {
 	return &jobManager{
-		client: client,
-		maker:  maker,
-		helper: helper,
-	}
-}
-
-func labels(mod kmmv1beta1.Module, targetKernel string) map[string]string {
-	return map[string]string{
-		constants.ModuleNameLabel:    mod.Name,
-		constants.TargetKernelTarget: targetKernel,
-		constants.JobType:            "build",
+		client:    client,
+		maker:     maker,
+		helper:    helper,
+		jobHelper: jobHelper,
 	}
 }
 
-func (jbm *jobManager) getJob(ctx context.Context, mod kmmv1beta1.Module, targetKernel string) (*batchv1.Job, error) {
-	jobList := batchv1.JobList{}
-
-	opts := []client.ListOption{
-		client.MatchingLabels(labels(mod, targetKernel)),
-		client.InNamespace(mod.Namespace),
-	}
-
-	if err := jbm.client.List(ctx, &jobList, opts...); err != nil {
-		return nil, fmt.Errorf("could not list jobs: %v", err)
+func (jbm *jobManager) GarbageCollect(ctx context.Context, mod kmmv1beta1.Module) ([]string, error) {
+	jobs, err := jbm.jobHelper.GetModuleJobs(ctx, mod, utils.JobTypeBuild)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get build jobs for module %s: %v", mod.Name, err)
 	}
 
-	if n := len(jobList.Items); n == 0 {
-		return nil, errNoMatchingBuild
-	} else if n > 1 {
-		return nil, fmt.Errorf("expected 0 or 1 job, got %d", n)
+	deleteNames := make([]string, 0, len(jobs))
+	for _, job := range jobs {
+		if job.Status.Succeeded == 1 {
+			opts := []client.DeleteOption{
+				client.PropagationPolicy(metav1.DeletePropagationBackground),
+			}
+			err = jbm.client.Delete(ctx, &job, opts...)
```

Comment:
Why isn't this using `DeleteJob` from the helper?

Reply:
fixed